diff --color -uNr a/fs/proc/cmdline.c b/fs/proc/cmdline.c
--- a/fs/proc/cmdline.c	2025-10-13 09:29:06.000000000 +0800
+++ b/fs/proc/cmdline.c	2025-10-13 17:53:57.255857716 +0800
@@ -27,11 +27,21 @@
 }
 #endif
 
+#ifdef CONFIG_KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+extern int susfs_spoof_cmdline_or_bootconfig(struct seq_file *m);
+#endif
+
 static int cmdline_proc_show(struct seq_file *m, void *v)
 {
 #ifdef CONFIG_INITRAMFS_IGNORE_SKIP_FLAG
 	seq_puts(m, proc_command_line);
 #else
+#ifdef CONFIG_KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+	if (!susfs_spoof_cmdline_or_bootconfig(m)) {
+		seq_putc(m, '\n');
+		return 0;
+	}
+#endif
 	seq_puts(m, saved_command_line);
 #endif
 	seq_putc(m, '\n');
