diff --color -uNr a/fs/namespace.c b/fs/namespace.c
--- a/fs/namespace.c	2025-10-29 10:21:14.000000000 +0800
+++ b/fs/namespace.c	2025-10-29 19:59:22.000000000 +0800
@@ -379,6 +379,10 @@
 		mnt->mnt_writers = 0;
 #endif
 		mnt->mnt.data = NULL;
+#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
+	// Make sure mnt->mnt.susfs_mnt_id_backup is initialized every time.
+	mnt->mnt.susfs_mnt_id_backup = 0;
+#endif
 
 		INIT_HLIST_NODE(&mnt->mnt_hash);
 		INIT_LIST_HEAD(&mnt->mnt_child);
@@ -1127,15 +1131,21 @@
 {
 	struct mount *mnt;
 	struct dentry *root;
-#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
-	// Make sure mnt->mnt.susfs_mnt_id_backup is initialized every time.
-	mnt->mnt.susfs_mnt_id_backup = 0;
-#endif
-	
+
 	if (!type)
 		return ERR_PTR(-ENODEV);
 
-	mnt = alloc_vfsmnt(name);
+#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
+    // We only check for ksu process
+        if (susfs_is_current_ksu_domain()) {
+                mnt = susfs_alloc_sus_vfsmnt(name);
+                goto bypass_orig_flow;
+        }
+#endif
+         mnt = alloc_vfsmnt(name);
+#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
+bypass_orig_flow:
+#endif
 
 	if (!mnt)
 		return ERR_PTR(-ENOMEM);